urlfc1c29991b_nil_mish_InJava_tgz-pJ8-edu_usc_cs_nsl_indexing_test_TestTrieProtoJ8/src/edu/usc/cs/nsl/indexing/QueryServer.java

```diff
@@ -0,0 +1,88 @@
+package edu.usc.cs.nsl.indexing;
+
+import edu.usc.cs.nsl.indexing.TrieDS.*;
+import edu.usc.cs.nsl.indexing.TrieDS.TrieProto.TriePB;
+
+import java.io.*;
+import java.util.*;
+
+/**
+ * Simple query server to take query from console.
+ * De-serializes a trie data structure from input file.
+ * @author nilmish
+ *
+ */
+
+public class QueryServer {
+
+	/**
+	 * @param args
+	 */
+	
+	private Trie the_trie;
+	
+	/**
+	 * Constructor for the query server which takes as input the location of the file which has the trie protocol buffer.
+	 * @param inputFile
+	 * @throws FileNotFoundException
+	 * @throws IOException
+	 */
+	public QueryServer(String inputFile) throws FileNotFoundException, IOException{
+		TriePB tp = TriePB.parseFrom(new FileInputStream(inputFile));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the resource created by the following constructor: *FileInputStream*. Currently, there are execution paths that do not contain closure statements, for example, when TriePB.parseFrom throws an exception. Ensure the object returned by *FileInputStream()* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).